src/MultiBridgeMessageSender.sol

```diff
@@ -253,7 +264,12 @@ contract MultiBridgeMessageSender {
             block.timestamp + _args.expiration
         );
         bytes32 msgId = MessageLibrary.computeMsgId(message);
-        bool[] memory adapterSuccess = _dispatchMessages(adapters, mmaReceiver, _args.dstChainId, message, _args.fees);
+        (bool[] memory adapterSuccess, uint256 successCount) =
+            _dispatchMessages(adapters, mmaReceiver, _args.dstChainId, message, _args.fees);
+
+        if (successCount < _args.successThreshold) {
```

Comment:
we can additionally also perform a check after the `_checkAndProcessArgs()` call to see if `_args.successThreshold > adapters.length` and fail earlier if so.